### PR TITLE
set storefront variant id when variant id is set + fixes

### PR DESCRIPTION
**Pull request by harisaurus**
- normalize variant ids when present, and use normalized variant id instead of database key.
- a bunch of normalizers
- tiny fixes

### CODE REVIEW

File: src/components/product.js [hunk 1]
```diff
@@ -52,11 +53,11 @@ function whitelistedProperties(selectorStyles) {
   }, {});
 }
 
-function normalizeProductId(config) {
-  if (config.storefrontId) {
-    return config.storefrontId;
-  } else if (config.id) {
-    return btoa(`gid://shopify/Product/${config.id}`);
+function normalizeId(type, config, databaseKey, storefrontKey) {
```
Comment (minasmart) [inline src/components/product.js]: Is this still used? OR was it moved to `normalizeConfig`?

File: src/components/product.js [hunk 2]
```diff
@@ -725,11 +727,11 @@ export default class Product extends Component {
    */
   setDefaultVariant(model) {
     let selectedVariant;
+    if (this.defaultStorefrontVariantId) {
+      selectedVariant = model.variants.find((variant) => btoa(`gid://shopify/ProductVariant/${variant.id}`) === this.defaultStorefrontVariantId);
```
Comment (minasmart) [inline src/components/product.js]: maybe export `normalizeId` from `normalizeConfig` too, so you can share logic

File: test/unit/product.js [hunk 3]
```diff
@@ -430,29 +430,41 @@ describe('Product class', () => {
...
     });
 
     it('calls init if variant ID updated', () => {
       product.updateConfig({variantId: 7777});
       assert.calledOnce(initSpy);
-      assert.equal(product.defaultVariantId, 7777);
+      assert.equal(product.defaultStorefrontVariantId, 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC83Nzc3');
+    });
+
+    it('calls init if storefront variant ID updated', () => {
```
Comment (minasmart) [inline test/unit/product.js]: so glad to see this.

File: src/components/product.js (hunk 2 above)
Comment (harisaurus) [reply src/components/product.js]: `normalizeId` is a little more complex than this scenario - it manages variantId when on a config object, whereas this context has no `config`.

might need to rework the normalizers to be a little more reusable.

**Pull request merged**